src/vectorial_query.py

```diff
@@ -0,0 +1,154 @@
+from config import WEIGHT_DOCUMENT, WEIGHT_QUERY, PATH_STOP_WORDS
+from collections import Counter
+import math
+from preprocess import InvertedIndex, StatCollection, lemmatize_document, tokenize_document, load_stop_words, remove_stop_words_from_document
+from typing import Optional, Dict, List, Union, Tuple
+
+def lemmatize_query(query: str) -> List[str]:
+    tokens = tokenize_document(query)
+    stop_words = load_stop_words(PATH_STOP_WORDS)
+    tokens = remove_stop_words_from_document(tokens, stop_words, [])
+    tokens = lemmatize_document(tokens)
+    return tokens
+
+def get_scores(
+    query: List[str], 
+    inverted_index: InvertedIndex, 
+    weighting_scheme_document=WEIGHT_DOCUMENT, 
+    weighting_scheme_query=WEIGHT_QUERY) -> Dict[int,float]:
+    """
+    compute score for all documents using the vectorial model with the config parameters
+    
+    Arguments:
+        query {List[str]} -- preprocessed query 
+        inverted_index {InvertedIndex} -- inverted index constructed on the collection
+    
+    Keyword Arguments:
+        weighting_scheme_document {str} -- weighting scheme for the document (default: {WEIGHT_DOCUMENT})
+        weighting_scheme_query {str} -- weighting scheme for the query (default: {WEIGHT_QUERY})
+    
+    Returns:
+        Dict[int,float] -- similarity between the query and each document
+    """
+    scores = {}
+    document_norms = {}
+    query_norm = 0
+    words = Counter(query)
+    stats_collection = inverted_index.stats
+    frequency_index = inverted_index.index
```

Comment:
Add maybe a `assert inverted_index.itype == 2`, because we're using a frequency index here